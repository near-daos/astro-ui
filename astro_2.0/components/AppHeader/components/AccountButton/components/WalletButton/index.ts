export * from 'astro_2.0/components/AppHeader/components/AccountButton/components/WalletButton/WalletButton';
