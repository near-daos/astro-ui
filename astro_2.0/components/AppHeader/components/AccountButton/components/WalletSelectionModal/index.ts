export * from 'astro_2.0/components/AppHeader/components/AccountButton/components/WalletSelectionModal/WalletSelectionModal';
