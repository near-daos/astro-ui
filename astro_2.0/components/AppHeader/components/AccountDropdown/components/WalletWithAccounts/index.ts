export * from 'astro_2.0/components/AppHeader/components/AccountDropdown/components/WalletWithAccounts/WalletWithAccounts';
