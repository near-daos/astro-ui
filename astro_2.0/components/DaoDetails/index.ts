export * from './DaoDetailsMinimized';
export * from './DaoDetailsGrid';
