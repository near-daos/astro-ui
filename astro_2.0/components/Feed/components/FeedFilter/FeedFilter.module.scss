@import 'styles/mixins';

.root {
  align-items: center;
  border: 1px solid var(--color-passive-gray);
  border-radius: 5px;
  display: inline-flex;
  height: 40px;
  justify-content: center;
  padding: 0 27px 0 20px;
}

.title,
.selectedTitle {
  color: var(--color-main-gray);
  display: inline-block;
  font-family: var(--font-family-secondary);
  font-size: 14px;
  line-height: 19px;
  margin: 0 12px 0 8px;
}

.selectedTitle {
  color: var(--color-black);
  display: none;
}

.radioGroup {
  align-items: center;
  display: flex;
}

.radio {
  margin: 0 8px;
  white-space: nowrap;
}

.header {
  align-items: center;
  display: flex;
  justify-content: flex-start;
  pointer-events: none;
  white-space: nowrap;
}

.controlIcon {
  display: none;
}

.toggle {
  margin-left: 15px;
}

.collapsed {
  &.root {
    align-items: stretch;
    padding: 0;
    position: relative;
  }

  .header {
    padding: 0 8px;
    pointer-events: auto;
    width: 100%;
  }

  .controlIcon {
    color: var(--color-black);
    cursor: pointer;
    display: inline-block;
    height: 24px;
    width: 24px;
  }

  .radioGroup {
    align-items: flex-start;
    background-color: #fff;
    border-radius: 8px;
    box-shadow: 3px 2px 24px rgba(0, 0, 0, 0.1);
    flex-direction: column;
    margin-top: 6px;
    min-width: 100%;
    opacity: 0;
    padding: 8px;
    pointer-events: none;
    position: absolute;
    right: 0;
    top: 100%;
    top: calc(100% - 10px);
  }

  .radio {
    height: 40px;
    margin: 0 !important;
    padding: 0 15px 0 10px;
    width: 100%;
  }

  .show {
    opacity: 1;
    pointer-events: auto;
    top: 100%;
    transition: all 0.3s ease-in-out;
    z-index: 100;
  }

  .activeRadio {
    background-color: var(--color-neutral-10);
  }

  .title {
    display: none;
  }

  .selectedTitle {
    display: inline-block;
  }
}
