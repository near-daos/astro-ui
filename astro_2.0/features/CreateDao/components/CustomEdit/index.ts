import CustomEdit from './CustomEdit';

export default CustomEdit;
