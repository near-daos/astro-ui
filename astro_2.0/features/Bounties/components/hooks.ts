import { useCallback, useRef, useState } from 'react';
import { useRouter } from 'next/router';
import { useAsyncFn, useLocalStorage, useLocation } from 'react-use';
import axios, { CancelTokenSource } from 'axios';

import { SputnikHttpService } from 'services/sputnik';
import { useModal } from 'components/modal';
import { ConfirmActionModal } from 'astro_2.0/components/ConfirmActionModal';

import { Bounty, BountyContext, BountyProposal } from 'types/bounties';
import { VoteAction } from 'types/proposal';

import { NOTIFICATION_TYPES, showNotification } from 'features/notifications';

import { useWalletContext } from 'context/WalletContext';
import { PaginationResponse } from 'types/api';
import { VOTE_ACTION_SOURCE_PAGE } from 'constants/votingConstants';

export function useBountyControls(
  daoId: string,
  bounty?: Bounty
): {
  handleClaim: () => void;
  handleUnclaim: () => void;
  loading: boolean;
} {
  const router = useRouter();
  const { nearService } = useWalletContext();
  const [, setVoteActionSource] = useLocalStorage(VOTE_ACTION_SOURCE_PAGE);
  const { pathname } = useLocation();
  const [loading, setLoading] = useState(false);

  const [showModal] = useModal(ConfirmActionModal);

  const onSuccessHandler = useCallback(async () => {
    router.reload();
  }, [router]);

  const handleClaim = useCallback(async () => {
    const res = await showModal({
      title: 'Confirm Your Claim',
    });

    if (res?.length && bounty) {
      setLoading(true);

      const dao = await SputnikHttpService.getDaoById(daoId);

      if (!dao) {
        return;
      }

      try {
        setVoteActionSource(pathname);

        await nearService?.claimBounty(daoId, {
          bountyId: bounty?.bountyId,
          deadline: bounty?.maxDeadline,
          bountyBond: dao.policy.bountyBond,
          gas: res[0],
          tokenId: bounty?.token,
        });

        await onSuccessHandler();
      } catch (err) {
        setLoading(false);
        showNotification({
          type: NOTIFICATION_TYPES.ERROR,
          lifetime: 5000,
          description: err.message,
        });
      }
    }
  }, [
    showModal,
    bounty,
    daoId,
    setVoteActionSource,
    pathname,
    nearService,
    onSuccessHandler,
  ]);

  const handleUnclaim = useCallback(async () => {
    const res = await showModal({
      title: 'Confirm Your Unclaim',
    });

    if (res?.length && bounty) {
      try {
        setLoading(true);
        setVoteActionSource(pathname);
        await nearService?.unclaimBounty(daoId, bounty?.bountyId);

        await onSuccessHandler();
      } catch (err) {
        setLoading(false);
        showNotification({
          type: NOTIFICATION_TYPES.ERROR,
          lifetime: 5000,
          description: err.message,
        });
      }
    }
  }, [
    showModal,
    bounty,
    setVoteActionSource,
    pathname,
    nearService,
    daoId,
    onSuccessHandler,
  ]);

  return {
    handleClaim,
    handleUnclaim,
    loading,
  };
}

export function useBountyVoting(
  daoId: string,
  proposal: BountyProposal
): {
  handleVote: (vote: VoteAction) => void;
  loading: boolean;
} {
  const router = useRouter();
  const [showModal] = useModal(ConfirmActionModal);
  const { nearService } = useWalletContext();
  const [, setVoteActionSource] = useLocalStorage(VOTE_ACTION_SOURCE_PAGE);
  const { pathname } = useLocation();

  const [{ loading }, handleVote] = useAsyncFn(
    async (vote: VoteAction) => {
      const res = await showModal({
        title: 'Confirm Your Vote',
      });

      if (res?.length) {
        setVoteActionSource(pathname);
        await nearService?.vote(daoId, proposal.proposalId, vote, res[0]);
        await router.reload();
      }
    },
    [daoId, proposal, router, nearService, pathname, setVoteActionSource]
  );

  return {
    handleVote,
    loading,
  };
}

export function useBountySearch(): {
  handleSearch: (
    val: string
  ) => Promise<PaginationResponse<BountyContext[]> | null>;
  loading: boolean;
} {
  const router = useRouter();
  const daoId = router.query.dao as string;
  const { accountId } = useWalletContext();
  const cancelTokenRef = useRef<CancelTokenSource | null>(null);

  const [{ loading }, handleSearch] = useAsyncFn(async query => {
    if (cancelTokenRef.current) {
      cancelTokenRef.current?.cancel('Cancelled by new req');
    }

    const { CancelToken } = axios;
    const source = CancelToken.source();

    cancelTokenRef.current = source;

    return SputnikHttpService.findBountyContext({
      daoId: daoId || '',
      accountId,
      query,
      cancelToken: source.token,
    });
  }, []);

  return {
    handleSearch,
    loading,
  };
}

export function useHideBounty(): {
  handleSubmit: () => void;
  handleSelect: (id: string) => void;
  selected: string[];
  loading: boolean;
} {
  const router = useRouter();
  const showHidden = router.query?.bountyFilter === 'hidden';
  const daoId = router.query.dao as string;
  const { accountId, nearService, pkAndSignature } = useWalletContext();
  const [selected, setSelected] = useState<string[]>([]);

  const [{ loading }, handleSubmit] = useAsyncFn(async () => {
    if (!pkAndSignature) {
      return;
    }

    try {
      const { publicKey, signature } = pkAndSignature;

      if (publicKey && signature) {
        await SputnikHttpService.toggleBountyContexts({
          accountId,
          publicKey,
          signature,
          daoId,
          ids: selected,
          isArchived: !showHidden,
        });

        await router.replace(router.asPath, undefined, {
          shallow: false,
        });

        setSelected([]);
      }
    } catch (err) {
      showNotification({
        type: NOTIFICATION_TYPES.ERROR,
        lifetime: 20000,
        description: err.message,
      });
    }
  }, [
    selected,
    daoId,
    showHidden,
    accountId,
    router,
    nearService,
    pkAndSignature,
  ]);

  const handleSelect = useCallback(
    (id: string) => {
      if (selected.includes(id)) {
        setSelected(selected.filter(item => item !== id));
      } else {
        setSelected([...selected, id]);
      }
    },
    [selected]
  );

  return {
    handleSubmit,
    handleSelect,
    selected,
    loading,
  };
}
