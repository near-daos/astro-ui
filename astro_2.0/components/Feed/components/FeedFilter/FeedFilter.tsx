import cn from 'classnames';
import React, {
  MutableRefObject,
  ReactNode,
  useCallback,
  useRef,
  useState,
} from 'react';

import { PAGE_LAYOUT_ID } from 'constants/common';

import { Icon } from 'components/Icon';
import { RadioGroup } from 'astro_2.0/components/inputs/radio/RadioGroup';
import { Radio, RadioProps } from 'astro_2.0/components/inputs/radio/Radio';

import { getElementSize } from 'utils/getElementSize';
import { useWindowResize } from 'hooks/useWindowResize';

import styles from './FeedFilter.module.scss';

type FeedFilterProps<T> = {
  className?: string;
  headerClassName?: string;
  title: string;
  shortTitle?: string;
  children: React.ReactElement<RadioProps, typeof Radio>[];
  value: T;
  onChange: (value: T, e?: React.ChangeEvent<HTMLInputElement>) => void;
  neighbourRef?: MutableRefObject<HTMLElement | null>;
  selectedLabel: ReactNode | string;
};

export const FeedFilter = <T,>({
  title,
  value,
  onChange,
  children,
  className,
  shortTitle,
  neighbourRef,
  headerClassName,
  selectedLabel,
}: FeedFilterProps<T>): JSX.Element => {
  const rootRef = useRef(null);

  const [maxWidth, setMaxWidth] = useState(0);
  const [collapsed, setCollapsed] = useState<boolean | null>(null);

  const collapseFilterIfNecessary = useCallback(() => {
    const pageLayoutEl = document.getElementById(PAGE_LAYOUT_ID);

    const rootEl = rootRef?.current;
    const neighbourEl = neighbourRef?.current;

    if (pageLayoutEl && rootEl) {
      const { offsetWidth } = rootEl;
      const { width } = getElementSize(pageLayoutEl);

      if (offsetWidth > maxWidth) {
        setMaxWidth(offsetWidth);
      }

      if (neighbourEl) {
        const { widthWithMargin } = getElementSize(neighbourEl);

        setCollapsed(width < widthWithMargin + Math.max(offsetWidth, maxWidth));
      } else {
        setCollapsed(width < maxWidth);
      }
    }
  }, [maxWidth, neighbourRef, setMaxWidth, setCollapsed]);

  useWindowResize(collapseFilterIfNecessary);

  const [isShow, setIsShow] = useState(false);

  const handleChange = useCallback(
    (newValue: string, e?: React.ChangeEvent<HTMLInputElement>) => {
      onChange(newValue as unknown as T, e);

      if (isShow) {
        setIsShow(false);
      }
    },
    [isShow, onChange]
  );

  const rootClassName = cn(styles.root, className, {
    [styles.collapsed]: collapsed,
  });

  return (
    <div className={rootClassName} ref={rootRef}>
      {collapsed !== null && (
        <>
          <div
            role="button"
            tabIndex={0}
            className={cn(styles.header, headerClassName)}
            onClick={() => setIsShow(!isShow)}
            onKeyDown={() => setIsShow(!isShow)}
          >
            <p className={styles.title}>{shortTitle || title}</p>
            <p className={styles.selectedTitle}>{selectedLabel}</p>
            <Icon className={styles.controlIcon} name="listFilter" />
          </div>
          <RadioGroup
            className={cn(styles.radioGroup, { [styles.show]: isShow })}
            itemClassName={styles.radio}
            activeItemClassName={styles.activeRadio}
            value={(value as unknown as string) || ''}
            onChange={handleChange}
          >
            {children}
          </RadioGroup>
        </>
      )}
    </div>
  );
};
