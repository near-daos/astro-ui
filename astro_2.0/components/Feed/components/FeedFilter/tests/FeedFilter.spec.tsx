/* eslint-disable @typescript-eslint/ban-ts-comment */

import React, { useState } from 'react';
import { render } from 'jest/testUtils';

import { FeedFilter } from 'astro_2.0/components/Feed/components/FeedFilter';

jest.mock('react', () => {
  return {
    ...jest.requireActual('react'),
    useState: jest.fn(),
  };
});

describe('FeedFilter', () => {
  it('Should render component', () => {
    // @ts-ignore
    useState.mockImplementation(() => [false, () => 0]);

    const { container } = render(
      <FeedFilter
        title="Hello World!"
        value="value"
        onChange={() => 0}
        selectedLabel="Label"
      >
        <div>1</div>
        <div>2</div>
      </FeedFilter>
    );

    expect(container).toMatchSnapshot();
  });
});
