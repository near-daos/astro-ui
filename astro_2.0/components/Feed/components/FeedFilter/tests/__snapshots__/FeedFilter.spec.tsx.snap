// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedFilter Should render component 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="header"
      role="button"
      tabindex="0"
    >
      <p
        class="title"
      >
        Hello World!
      </p>
      <p
        class="selectedTitle"
      >
        Label
      </p>
      <svg
        class="controlIcon"
      >
        <use
          href="/_next/undefined"
        />
      </svg>
    </div>
    <div
      class="radioGroup"
      role="radiogroup"
    >
      <div>
        1
      </div>
      <div>
        2
      </div>
    </div>
  </div>
</div>
`;
